rasa_core/featurizers.py

```
@@ -163,18 +136,425 @@ def encode(self, active_features, input_feature_map):
+                _json = f.read()
+            return jsonpickle.decode(_json)
+        else:
+            logger.info("Couldn't load featurizer for policy. "
+                        "File '{}' doesn't exist. ".format(featurizer_file))
+            return None
+
+
+class FullDialogueFeaturizer(Featurizer):
+    def __init__(self, featurize_mechanism):
+        # type: (FeaturizeMechanism) -> None
+        super(FullDialogueFeaturizer, self).__init__(featurize_mechanism)
+
+        self.max_len = None
+
+    def _calculate_max_len(self, as_states):
+        self.max_len = 0
+        for states in as_states:
+            self.max_len = max(self.max_len, len(states))
+        logger.info("The longest dialogue has {} actions."
+                    "".format(self.max_len))
+
+    def _pad_states(self, states):
+        # pad up to max_len
+        if len(states) < self.max_len:
+            states += [None] * (self.max_len - len(states))
+
+        return states
+
+    def training_states_and_actions(
+            self,
+            trackers,  # type: List[DialogueStateTracker]
+            domain  # type: Domain
+    ):
+        # type: (...) -> Tuple[List[List[Dict]], List[List[Dict]], Dict]
+
+        trackers_as_states = []
+        trackers_as_actions = []
+        events_metadata = defaultdict(set)
+
+        for tracker in trackers:
+            states = domain.states_for_tracker_history(tracker)
+
+            delete_first_state = False
+            actions = []
+            for event in tracker._applied_events():
+                if isinstance(event, ActionExecuted):
+                    if not event.unpredictable:
+                        # only actions which can be predicted at a stories start
+                        actions.append(event.action_name)
+                    else:
+                        # unpredictable actions can be only the first in the story
+                        delete_first_state = True
```

let's raise an exception if `delete_first_state ` is already set to `True` (meaning that there are two actions that are unpredictable, which shouldn't really be the case)